lib/docker.ts

```diff
@@ -128,15 +130,43 @@ class Docker {
     exclusionGlobs: string[] = [],
     path: string = "/",
     recursive: boolean = true,
+    excludeRootDirectories: string[] = SystemDirectories,
   ) {
+    let root: lsu.DiscoveredDirectory;
     const res: string[] = [];
-    const output = await this.lsSafe(path, recursive);
 
-    if (eventLoopSpinner.isStarving()) {
-      await eventLoopSpinner.spin();
-    }
+    if (recursive && path === "/") {
+      // When scanning from the root of a docker image we need to
+      // exclude system files e.g. /proc, /sys, etc. to make the
+      // operation less expensive.
+
+      const outputRoot = await this.lsSafe("/", false);
```

Comment:
We do a non-recursive ls on the root and then for each subdirectory we do a recursive one except on the excluded system directories. We then merge the results together.